src/solver/cartesianprod/backward_cartesianprod.cpp

```diff
@@ -0,0 +1,208 @@
+                                                                 build_params));
+        }
+
+        result.invoker_factory = [inputCount](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+                uint64_t stride       = 1;
+                int kernelCnt         = 0;
+                decltype(auto) params = raw_params.CastTo<miopen::cartesianprod::BwdInvokeParams>();
+                auto output_grad_tv   = get_inner_expanded_tv<2>(deref(params.outputGradDesc));
+                HipEventPtr start, stop;
+                bool profiling = handle_.IsProfilingEnabled();
+                if(profiling)
+                {
+                    handle_.EnableProfiling(false);
+                    hipStreamSynchronize(handle_.GetStream());
+                    start = miopen::make_hip_event();
+                    stop  = miopen::make_hip_event();
+                    hipEventRecord(start.get(), handle_.GetStream());
+                }
+
+                for(int i = inputCount - 1; i >= 0; i--)
+                {
+                    decltype(auto) kernel = handle_.Run(kernels[kernelCnt++]);
+                    auto input_grad_tv =
+                        get_inner_expanded_tv<1>(deref(params.GetInputGradDesc(i)));
+                    kernel(params.output_grad,
+                           params.GetInputGrad(static_cast<uint64_t>(i)),
+                           output_grad_tv,
+                           input_grad_tv,
+                           stride,
+                           static_cast<uint64_t>(i));
+                    stride *= params.GetInputGradDesc(i)->GetElementSize();
+                }
+
+                if(profiling)
+                {
+                    float elapsed = 0.0f;
+                    hipEventRecord(stop.get(), handle_.GetStream());
+                    handle_.EnableProfiling(true);
+                    hipEventSynchronize(stop.get());
+                    hipEventElapsedTime(&elapsed, start.get(), stop.get());
+                    // Clean up
+                    hipEventDestroy(start.get());
+                    hipEventDestroy(stop.get());
+                    handle_.ResetKernelTime();
+                    handle_.AccumKernelTime(elapsed);
+                };
+            };
+        };
+    }
+    else
+    {
+        result.invoker_factory = [=](const std::vector<Kernel>&) {
+            return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+                HipEventPtr start, stop;
+                bool profiling = handle_.IsProfilingEnabled();
+                if(profiling)
+                {
+                    handle_.EnableProfiling(false);
+                    hipStreamSynchronize(handle_.GetStream());
+                    start = miopen::make_hip_event();
+                    stop  = miopen::make_hip_event();
+                    hipEventRecord(start.get(), handle_.GetStream());
+                }
```

Comment:
Do we really need this setup?

Reply:
hipMemcpy is not a kernel so elapsed_time_handler will not be active, if you not set up this way, profiler can not record the time 
<img width="881" alt="image" src="https://github.com/user-attachments/assets/978f70d8-491d-4303-889f-60a7bbc4dfc8">